html/captive-portal/lib/captiveportal/PacketFence/Model/Portal/Session.pm

```diff
@@ -92,6 +92,8 @@ has [qw(forwardedFor guestNodeMac)] => ( is => 'rw', );
 sub ACCEPT_CONTEXT {
     my ( $self, $c, @args ) = @_;
     my $class = ref $self || $self;
+    my $previous_model = $c->session->{$class};
+    return $previous_model if(defined($previous_model) && $previous_model->{options}->{in_uri_portal});
```

Comment:
Because in the event that we use last_uri as really the last_uri the user visited then that would break the code. Would seem like a misusage to my eyes.